SpliceAI.pm

```diff
@@ -0,0 +1,263 @@
+  }
+
+  return $self;
+}
+
+sub feature_types {
+  return ['Transcript'];
+}
+
+sub get_header_info {
+  my $self = shift;
+
+  my %header;
+  $header{'SpliceAI_pred'} = 'SpliceAI predicted effect on splicing. These include delta scores (DS) and delta positions (DP) for acceptor gain (AG), acceptor loss (AL), donor gain (DG), and donor loss (DL). Format: ALLELE:SYMBOL|DS_AG|DS_AL|DS_DG|DS_DL|DP_AG|DP_AL|DP_DG|DP_DL';
+
+  if($self->{cutoff}) {
+    $header{'SpliceAI_cutoff'} = 'Flag if delta score pass the cutoff (PASS) or if it does not (FAIL)';
+  }
+
+  return \%header;
+}
+
+sub run {
+  my ($self, $tva) = @_;
+  my $vf = $tva->variation_feature;
+  my $chr = $vf->{chr};
+
+  my $end = $vf->{end};
+  my $start = $vf->{start};
+  ($start, $end) = ($end, $start) if $start > $end;
+
+  my @data = @{$self->get_data($chr, $start, $end)} if(defined $chr);
+
+  return {} unless(@data);
+
+  my $result_data = '';
+  my $result_flag;
+
+  foreach my $data_value (@data) {
+
+    my $ref_allele;
+    my $alt_allele;
+
+    # get alt allele
+    my $allele = $tva->variation_feature_seq;
+    reverse_comp(\$allele) if $vf->{strand} < 0;
+    my $new_allele_string = $vf->ref_allele_string.'/'.$allele;
+
+    if($vf->ref_allele_string =~ /-/) {
+
+      # convert to vcf format to compare the alt alleles
+      my $vf_2 = Bio::EnsEMBL::Variation::VariationFeature->new
+        (-start => $start,
+         -end => $end,
+         -strand => $vf->{strand},
+         -allele_string => $new_allele_string
+        );
+      my $convert_to_vcf = $vf_2->to_VCF_record;
+      $ref_allele = ${$convert_to_vcf}[3];
+      $alt_allele = ${$convert_to_vcf}[4];
+    }
+    else {
+      $ref_allele = $vf->ref_allele_string;
+      $alt_allele = $allele;
+    }
+
+    my $matches = get_matched_variant_alleles(
+      {
+        ref    => $ref_allele,
+        alts   => [$alt_allele],
+        pos    => $start,
+        strand => $vf->strand
+      },
+      {
+       ref  => $data_value->{ref},
+       alts => [$data_value->{alt}],
+       pos  => $data_value->{start},
+      }
+    );
+    if (@$matches) {
+
+      my %hash;
+      $hash{'SpliceAI_pred'} = $data_value->{result};
+
+      # Add a flag if cutoff is used
+      if($self->{cutoff}) {
+        if($data_value->{info} >= $self->{cutoff}) {
+          $result_flag = 'PASS';
+        }
+        else {
+          $result_flag = 'FAIL';
+        }
+        $hash{'SpliceAI_cutoff'} = $result_flag;
+      }
+
+      return \%hash;
+    }
+  }
+
+  return {};
+}
+
+# Parse data from SpliceAI file
+sub parse_data {
+  my ($self, $line) = @_;
+
+  my ($chr, $start, $id, $ref, $alt, $qual, $filter, $info) = split /\t/, $line;
+
+  $info =~ s/SpliceAI=//;
+  my @info_splited = split (qr/\|/,$info, 2);
+  my $allele = $info_splited[0];
+  my $data = $info_splited[1];
+
+  my $max_score;
+  if($self->{cutoff}){
+    my @scores = split (qr/\|/,$data);
+
+    my @scores_list;
+    push @scores_list, $scores[1];
```

Comment:
Suggest using an array slice
@scores_list = @scores[1 .. 4]

Reply:
Thanks for the suggestion.